### ISSUE TITLE
elasticache_subnet_group does not return any facts

### ISSUE THREAD

**Issue by fperks** [ansible/ansible-modules-core#2299]
**Issue Type**: Bug Report
**Ansible Version**: ansible 2.0.0 (devel 5ae74f7d38) last updated 2015/10/06 09:16:13 (GMT -400)
**Ansible Configuration**: No 
**Environment**: Ubuntu 14.04
**Summary**:

Unfortunately the elasticache_subnet_group module does not actually return any facts about what it just created. This unfortunately causes an issue because i have no information about the subnet group i just created, so i can't pass it to cache_subnet_group on the elasticache module.

**Steps to Reproduce**:

```
- name: Deploying elasticache subnet groups
  elasticache_subnet_group:
    name: hats
    description: "my awesome elasticache"
    subnets: "{{ my_awesome_subnets }}"
    region: us-west-2
    state: present
  register: result_elasticache_subnet_group
```

**Expected Results**: I expect that the module would at least return a little bit of information about what it just created. So it can be used in future tasks. 

**Actual Results**: 

```
changed: [localhost] => {"changed": true}
```

Printing the result out with debug:

```
TASK [XXXXXXXX : debug var=result_elasticache_subnet_group] ***********
ok: [localhost] => {
    "changed": false,
    "result_elasticache_subnet_group": {
        "_ansible_no_log": false,
        "changed": true
    }
}
```

**Comment by robynbergeron** [comment]
[module: cloud/amazon/elasticache_subnet_group.py]

**Comment by robynbergeron** [comment]
Greetings, @fperks -- thanks for submitting this bug report.

@timmahoney -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by timmahoney** [comment]
Hi!

Unfortunately AWS doesn't assign IDs to EC subnet groups, like it does almost every other resource in their various services. In your example, your subnet group name would be "hats" and you would just assign 

```
cache_subnet_group: hats
```

More complete example:

```
- name: Maintain an Elasticache Subnet Group
  elasticache_subnet_group:
    name: "{{ subnet_group_name | default('hats') }}"
    description: "Description for your subnet group"
    subnets:
    - "subnet-id-xxx"
    - "subnet-id-yyy"
    - "subnet-id-zzz"
    region: {{ region }}
    state: present

- name: Maintain an Elasticache Memcached Cluster
  elasticache:
    name: "name-of-your-elasticache-cluster"
    state: present
    engine: "memcached"
    cache_engine_version: "1.4.14"
    node_type: "cache.t2.micro"
    num_nodes: "1"
    cache_port: "11211"
    cache_subnet_group: "{{ subnet_group_name | default('hats') }}"
    security_group_ids:
    - "sg-XXXXXX"
    region: {{ region }}
    wait: no
```

This module is based off of the RDS Subnet Group module, which behaves the same way, I imagine for the same reason.

**Comment by ansibot** [comment]
@timmahoney, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by timmahoney** [comment]
This issue can be closed.

**Comment by timmahoney** [comment]
wontfix This is a limitation in AWS.
